### PR TITLE
Improve ruff config

### PR DESCRIPTION
**Pull request by AlexWaygood**
- Add `fix = true` to `pyproject.toml`. This means ruff will apply autofixes when possible if `ruff .` is run (and will apply autofixes to PRs).
- Remove isort as a test dependency; just use ruff instead. It has the same import-sorting style, is faster, and requires less configuration.
- Remove an outdated paragraph in `CONTRIBUTING.md` about how you'll get a CI failure if you don't run black and isort before filing a PR. This is no longer true since we switched to using pre-commit.ci.

### CODE REVIEW

**Review by hauntsaninja — COMMENTED** [review]
If we're going to have fix on by default, can we mark the following as unfixable:
```
unfixable = [
    "F841",  # unused variable. ruff keeps the call, but mostly we want to get rid of it all
    "F601",  # automatic fix might obscure issue
    "F602",  # automatic fix might obscure issue
    "B018",  # automatic fix might obscure issue
]
```

**Comment by ikonst** [comment]
So I gather they "autofix" [B018](https://beta.ruff.rs/docs/rules/useless-expression/) ("Found useless expression. Either assign it to a variable or remove it.") by ... removing the expression rather than assigning it?

**Comment by AlexWaygood** [comment]
> Update: https://beta.ruff.rs/docs/rules/#flake8-bugbear-b doesn't suggest B018 has a fixer currently. Are you adding it just as a precaution?

I just went with Shantanu's suggestions here, but I definitely agree that it's good in general to be cautious with autofixes, and I agree that if they were to add an autofix for B018, we probably wouldn't want it

**Comment by hauntsaninja** [comment]
Oh hmmm I don't remember why I marked B018 unfixable, possibly just defensive. I added ruff to my work codebase a couple weeks ago and my suggestion^ is just whatever I'd decided was good at that time.

In case you're curious, this is what I have currently at work:
<details>

```
select = [
    "E", "F", "W", "B", "C4", "ASYNC", "PIE",
    # grab bag of random things
    "DTZ003", "DTZ004",
    "PLE1310", "PLW0120",
    "SIM101", "SIM110", "SIM201", "SIM202", "SIM222", "SIM223",
    "S506",
    "RET501", "RET502",  # RET503 is also nice, but there's currently some noise
    "RUF006", "RUF200",
    "COM819",
]
ignore = [
    "E402",   # people usually know what they're doing
    "E501",   # line too long. black does a good enough job
    "E731",   # lambda expression assignments. these are nice sometimes
    "E741",   # variable names like "l"
    "B011",   # assert false. we don't use python -O
    "C408",   # we often use dict with keyword args
]
unfixable = [
    "F841",  # unused variable. ruff keeps the call, but mostly we want to get rid of it all
    "F601",  # automatic fix might obscure issue
    "F602",  # automatic fix might obscure issue
    "B018",  # automatic fix might obscure issue
    "SIM222",  # automatic fix might obscure issue
    "SIM223",  # automatic fix might obscure issue
]
```
</details>

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -9,10 +9,6 @@ repos:
     rev: 23.3.0  # must match test-requirements.txt
     hooks:
       - id: black
-  - repo: https://github.com/pycqa/isort
-    rev: 5.12.0  # must match test-requirements.txt
-    hooks:
-      - id: isort
   - repo: https://github.com/astral-sh/ruff-pre-commit
     rev: v0.0.272  # must match test-requirements.txt
```
Comment (hauntsaninja) [inline .pre-commit-config.yaml:13]: I think will need to add something like `--exit-non-zero-on-fix` down here

**Review by hauntsaninja — CHANGES_REQUESTED** [review]

**Review by hauntsaninja — APPROVED** [review]

**Pull request merged**